File: 01_algorithm/sort/02_selection/selection_sort.dart
import '../../../00_util/dart/generate.dart';

List<int> selectionSort(List<int> nums){
  var l = nums.length;
  for(var i=0; i < l; i++){
    var tmpIndex = i;
    for(var j=i; j < l; j++){
      if (nums[j]< nums[tmpIndex]){
        tmpIndex = j;
      }
    }
    var tmp = nums[i];
    nums[i] = nums[tmpIndex];
    nums[tmpIndex] = tmp;
  }
  return nums;
}

void main(){
  var nums = [12, 11, 13, 5, 6, 7];
  selectionSort(nums);
  print(nums);

  List<int> testNums = GenerateRandomArray(6);
  selectionSort(testNums);
  print(testNums);
}

File: 01_algorithm/sort/02_selection/selection_sort.go
package main

import (
	util "coding/00_util/golang"
	"fmt"
)

func selectionSort(arr []int) []int {

	l := len(arr)
	for i := 0; i < l; i++ {
		tmpIndex := i
		for j := i; j < l; j++ {
			if arr[j] < arr[tmpIndex] {
				tmpIndex = j
			}
		}
		arr[i], arr[tmpIndex] = arr[tmpIndex], arr[i]
	}
	return arr
}

func main() {
	fmt.Println(selectionSort([]int{5, 4, 3, 2, 1}))
	fmt.Println(selectionSort(util.GenerateRandomArray(5)))
}
